crates/aptos/src/move_tool/transactional_tests_runner.rs

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+/// Most of the code below comes from the crate `datatest-stable`. Because the limitation of `datatest-stable`,
+/// we are not able to run transactional tests as a subcommand of the Aptos CLI. Therefore, we need to duplicate code
+/// here and make minor modifications.
+///
+use clap::Parser;
+
+use std::{
+    io::{self, Write},
+    num::NonZeroUsize,
+    panic::{catch_unwind, AssertUnwindSafe},
+    path::{Path, PathBuf},
+    process,
+    sync::mpsc::{channel, Sender},
+    thread,
+};
+
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+/// Run Move transactional tests
+#[derive(Parser, Clone)]
+pub struct TransactionalTestOpts {
+    /// The FILTER string is tested against the name of all tests, and only those tests whose names
+    /// contain the filter are run.
+    #[clap(long)]
+    pub filter: Option<String>,
+
+    /// Exactly match filters rather than by substring
+    #[clap(long = "exact")]
+    pub filter_exact: bool,
+
+    /// Number of threads used for running tests in parallel
+    #[clap(long, default_value = "32")]
+    pub test_threads: NonZeroUsize,
+
+    /// Output minimal information
+    #[clap(short = 'q', long)]
+    pub quiet: bool,
+
+    /// List all tests
+    #[clap(long)]
+    pub list: bool,
+
+    /// Path to contain the tests
+    #[clap(long, parse(from_os_str))]
+    pub root_path: PathBuf,
+
+    /// Pattern to match the test files
+    #[clap(long, default_value = r".*\.(mvir|move)$")]
+    pub pattern: String,
+}
+
+/// Helper function to iterate through all the files in the given directory, skipping hidden files,
+/// and return an iterator of their paths.
+pub fn iterate_directory(path: &Path) -> impl Iterator<Item = PathBuf> {
+    walkdir::WalkDir::new(path)
+        .into_iter()
+        .map(::std::result::Result::unwrap)
```

Comment:
Let's try to avoid `unwrap()` and return CliError so that we have a consistent error system.